client/windows/CMessage.cpp

```diff
@@ -60,125 +60,227 @@ void CMessage::dispose()
 		item.reset();
 }
 
-std::vector<std::string> CMessage::breakText(std::string text, size_t maxLineWidth, EFonts font)
+std::vector<size_t> CMessage::getAllPositionsInStr(const char & targetChar, const std::string_view & str)
 {
-	assert(maxLineWidth != 0);
-	if(maxLineWidth == 0)
-		return {text};
-
-	std::vector<std::string> ret;
-
-	boost::algorithm::trim_right_if(text, boost::algorithm::is_any_of(std::string(" ")));
+	std::vector<size_t> results;
+	for(size_t pos = 0; pos < str.size(); pos++)
+	{
+		if(str[pos] == targetChar)
+			results.push_back(pos);
+	}
+	return results;
+};
 
-	const auto & fontPtr = GH.renderHandler().loadFont(font);
+bool CMessage::validateColorBraces(const std::string & text)
+{
+	const auto openingBraces = getAllPositionsInStr('{', text);
+	const auto closingBraces = getAllPositionsInStr('}', text);
 
-	// each iteration generates one output line
-	while(text.length())
+	if(openingBraces.size() != closingBraces.size())
 	{
-		ui32 wordBreak = -1; //last position for line break (last space character)
-		ui32 currPos = 0; //current position in text
-		bool opened = false; //set to true when opening brace is found
-		std::string color; //color found
-
-		size_t symbolSize = 0; // width of character, in bytes
+		logGlobal->error("CMessage: Not Equal number opening and closing tags");
+		return false;
+	}
 
-		std::string printableString;
+	if(openingBraces.empty())
+		return true;
 
-		// loops till line is full or end of text reached
-		while(currPos < text.length() && text[currPos] != 0x0a && fontPtr->getStringWidth(printableString) <= maxLineWidth)
+	auto closingPos = closingBraces.begin();
+	size_t closingPrevPos = 0;
+	for(const auto & openingPos : openingBraces)
+	{
+		if(openingPos >= *closingPos)
 		{
-			symbolSize = TextOperations::getUnicodeCharacterSize(text[currPos]);
-
-			// candidate for line break
-			if(ui8(text[currPos]) <= ui8(' '))
-				wordBreak = currPos;
+			logGlobal->error("CMessage: Line validation Invalid: Closing tag before opening tag %s", text);
+			return false;
+		}
+		if(openingPos < closingPrevPos)
+		{
+			logGlobal->error("CMessage: Line validation Invalid: Tags inside another tags %s", text);
+			return false;
+		}
+		closingPrevPos = *closingPos++;
+	}
+	return true;
+}
 
-			/* We don't count braces in string length. */
-			if(text[currPos] == '{')
+std::vector<CMessage::coloredline> CMessage::getPossibleLines(
+	const std::string & line, size_t maxLineWidth, const EFonts font, const char & splitSymbol)
+{
+	const auto & fontPtr = GH.renderHandler().loadFont(font);
+	std::vector<coloredline> result;
+	std::string_view lastFoundColor;
+	const std::function<size_t(const std::string_view&)> findColorTags = [&](const std::string_view & subLine) -> size_t
+		{
```

Comment:
Might be better to have them as separate method. Large lambdas inside method are hard to read

Reply:
Done